mlir/lib/Dialect/Linalg/Transforms/DataLayoutPropagation.cpp

```diff
@@ -63,6 +63,13 @@ getPackingInfoFromOperand(OpOperand *opOperand, linalg::GenericOp genericOp,
                           OpTy packOrUnPackOp) {
   static_assert(llvm::is_one_of<OpTy, linalg::PackOp, linalg::UnPackOp>::value,
                 "applies to only pack or unpack operations");
+  // TODO(issues/129004): Support MemRef PackOp. Temporarily return failure.
+  if (auto linalgOp =
+          dyn_cast<linalg::LinalgOp>(packOrUnPackOp.getOperation())) {
```

Comment:
The condition will always be false because pack/unpack does not implement LinalgOp interface. I think it is unclear to do layout propagation on memrefs because there are memory scope ish things. So maybe drop the TODO and just return failure when it does not have pure tensor semantics.

EDIT: we don't need the change in the local function because we should already bail out if they are on memrefs. This is dead code.